src/gwt/src/org/rstudio/studio/client/rsconnect/RSConnect.java

```diff
@@ -800,6 +821,17 @@ private void uploadToRPubs(RSConnectPublishInput input,
          RSConnectPublishResult result,
          final ProgressIndicator indicator)
    {
+      if (input.getContentType() == CONTENT_TYPE_DOCUMENT && 
+            (!input.hasDocOutput() || !supportedRPubsDocExtension(input.getDocOutput())))
+      {
+         display_.showErrorMessage("Publish Document",
```

Comment:
Nit: as earlier, I think we should have a separate error message for unsupported extensions